### PR TITLE
feat(interchain-indexer): Stats zero-count chain visibility setting

### PR DESCRIPTION
**Pull request by EvgenKor**
## Summary

- Add a service-wide `stats.include_zero_chains` setting, defaulting to `true`, so stats reads can either include known chains with zero-valued data or keep the previous stats-only behavior.
- Make `/api/v1/stats/chains` honor that toggle without breaking SQL-side ordering, filtering, or keyset pagination.
- Extend sent/received message-path endpoints to synthesize zero-count counterparties from `chains` only when include-zero mode is enabled and no explicit `counterparty_chain_ids` filter is provided.
- Keep the API contract unchanged while preserving message-path invariants: no synthetic self-edge for the selected chain and no synthetic rows for explicitly requested counterparties.

## Implementation Plan

- Group stats runtime settings under a nested `StatsSettings` struct in server config and thread the read toggle into `StatsService` via `StatsReadSettings`.
- Route message-path reads through `StatsService` so the zero-chain visibility rule is owned by the stats subsystem instead of being split across server handlers and raw DB calls.
- Update `stats_chains_query.rs` to accept `include_zero_chains` and apply `COALESCE(..., 0) > 0` inside the ranked SQL when zero rows should be omitted.
- Add toggle-aware message-path SQL branches in `database.rs`:
  - keep the existing stats-table queries in omit-zero mode
  - expand from `chains` with `LEFT JOIN` / `COALESCE` in include-zero mode
  - preserve all-time and bounded-date parity, deterministic ordering, self-edge exclusion, and counterparty short-circuit behavior
- Update config examples and constructor call sites to use the new nested stats settings shape and default read settings.
- Add DB-backed test coverage for include-zero and omit-zero behavior in `/stats/chains` and message-path queries.

## API Changes

- None. Request and response contracts are unchanged; the difference is limited to which rows the existing stats endpoints return.

## ENV / Config Changes

- Introduce nested stats settings:
  - `stats.backfill_on_start`
  - `stats.chains_recalculation_period_secs`
  - `stats.include_zero_chains`
- Introduce / use the corresponding env vars:
  - `INTERCHAIN_INDEXER__STATS__BACKFILL_ON_START`
  - `INTERCHAIN_INDEXER__STATS__CHAINS_RECALCULATION_PERIOD_SECS`
  - `INTERCHAIN_INDEXER__STATS__INCLUDE_ZERO_CHAINS`
- Default `stats.include_zero_chains` to `true`, so the shipped behavior is to return known chains even when stats rows are missing or zero.
- Replace the old flat stats config shape with the nested `stats.*` block. Because settings use `deny_unknown_fields`, stale flat keys will fail validation rather than being ignored.

## Database / Migration Impact

- None. No schema, migration, backfill format, or projection-table changes are introduced.

## Testing

- Recorded verification:
  - `cargo check`
  - `cargo clippy --all --all-targets --all-features`
  - `just test settings`
  - `just check-envs`
  - `just test-with-db`
- Added ignored DB-backed tests covering:
  - `/stats/chains` include-zero and omit-zero behavior
  - pagination with zero-row filtering
  - outgoing / incoming message-path expansion in include-zero mode
  - bounded-date parity
  - self-edge exclusion
  - counterparty filters not synthesizing zero rows
  - omit-zero mode preserving stats-only behavior

## Summary by CodeRabbit

* **Configuration**
  * Stats settings reorganized from top-level to nested `[stats]` configuration section
  * Updated environment variable naming with double underscores (e.g., `INTERCHAIN_INDEXER__STATS__BACKFILL_ON_START`)
  * Added `include_zero_chains` configuration flag (default `true`) to control stats endpoint behavior when data is missing or zero

### CODE REVIEW

File: interchain-indexer/interchain-indexer-server/src/server.rs [hunk 1]
```diff
@@ -118,9 +118,12 @@ pub async fn run(settings: Settings) -> Result<(), anyhow::Error> {
...
         Some(token_info_service.clone()),
+        StatsReadSettings {
+            include_zero_chains: settings.stats.include_zero_chains,
+        },
     ));
 
-    if settings.stats_backfill_on_start {
+    if settings.stats.backfill_on_start {
         tracing::info!(
             "stats_backfill_on_start enabled: running statistics projection; async token enrichment will run after each batch outside DB transactions"
```
[the remark this answers is not in the document]
Comment (EvgenKor) [reply interchain-indexer/interchain-indexer-server/src/server.rs]: Fixed in https://github.com/blockscout/blockscout-rs/pull/1662/commits/dcfbee7fe9a77928b096d6d55db3e4ea4c6c4e6e

File: interchain-indexer/justfile [hunk 2]
```diff
@@ -36,7 +36,6 @@ stop-test-postgres:
 run:
     INTERCHAIN_INDEXER__DATABASE__CONNECT__URL={{DATABASE_URL}} \
     INTERCHAIN_INDEXER__DATABASE__RUN_MIGRATIONS=true \
-    INTERCHAIN_INDEXER__STATS_BACKFILL_ON_START=true \
     INTERCHAIN_INDEXER__CHAINS_CONFIG=config/avalanche/chains.json \
     INTERCHAIN_INDEXER__BRIDGES_CONFIG=config/avalanche/bridges.json \
     INTERCHAIN_INDEXER__API__USE_PAGINATION_TOKEN=true \
```
[the remark this answers is not in the document]
Comment (EvgenKor) [reply interchain-indexer/justfile:41]: This was one-time ENV. It doesn't need anymore, so it was deleted.

File: interchain-indexer/interchain-indexer-logic/src/stats_chains_query.rs [hunk 3]
```diff
@@ -162,39 +163,36 @@ pub async fn list_stats_chains(
...
+        inner_conditions.push(format!(
+            "(c.name ILIKE ${ph} ESCAPE '\\' OR CAST(c.id AS TEXT) ILIKE ${ph} ESCAPE '\\')",
+        ));
         values.push(Value::String(Some(Box::new(pat))));
-        clause
-    } else {
+    }
+
+    if !include_zero_chains {
+        inner_conditions.push("COALESCE(sc.unique_transfer_users_count, 0) > 0".to_string());
```
[the remark this answers is not in the document]
Comment (EvgenKor) [reply interchain-indexer/interchain-indexer-logic/src/stats_chains_query.rs]: Agree. Fixed in https://github.com/blockscout/blockscout-rs/pull/1662/commits/dcfbee7fe9a77928b096d6d55db3e4ea4c6c4e6e

**Comment by EvgenKor** [comment]
@coderabbit review

File: interchain-indexer/interchain-indexer-logic/src/stats_chains_query.rs [hunk 4]
```diff
@@ -162,39 +163,39 @@ pub async fn list_stats_chains(
...
         values.push(Value::String(Some(Box::new(pat))));
-        clause
-    } else {
+    }
+
+    if !include_zero_chains {
+        inner_conditions.push(
+            "(COALESCE(sc.unique_transfer_users_count, 0) > 0 OR COALESCE(sc.unique_message_users_count, 0) > 0)"
+                .to_string(),
+        );
```
[the remark this answers is not in the document]
Comment (EvgenKor) [reply interchain-indexer/interchain-indexer-logic/src/stats_chains_query.rs:190]: This in intended behaviour. We should skip only chains without any activity. In future `unique_message_users_count` most likely will be added to the chain stats

File: interchain-indexer/README.md [hunk 5]
```diff
@@ -102,8 +102,9 @@ Defines which bridges (cross-chain mechanisms) to index. Each entry is one bridg
...
+| `INTERCHAIN_INDEXER__STATS__CHAINS_RECALCULATION_PERIOD_SECS`           |                          | Interval in seconds between full recomputations of per-chain distinct user counters in `stats_chains` (from `crosschain_messages` / `crosschain_transfers`, any status). Only chains with at least one counted user address keep a row; stale rows are deleted. Set to `0` to disable the background task. | `3600`        |
+| `INTERCHAIN_INDEXER__STATS__INCLUDE_ZERO_CHAINS`                        |                          | When `true`, stats endpoints include known chains from `chains` even when the aggregated stats row is missing or has a zero value. Disable it to return only chains with positive aggregated stats. | `true`        |
```
[the remark this answers is not in the document]
Comment (EvgenKor) [reply interchain-indexer/README.md]: Fixed in https://github.com/blockscout/blockscout-rs/pull/1662/commits/3a445f9e498cd928c8446298e732abae80949629

File: interchain-indexer/interchain-indexer-logic/src/database.rs [hunk 6]
```diff
@@ -173,11 +173,55 @@ enum MessagePathDirection {
...
+LEFT JOIN stats_messages sm
+    ON sm.src_chain_id = c.id
+   AND sm.dst_chain_id = $1
+WHERE c.id <> $1
+ORDER BY messages_count DESC, src_chain_id ASC, dst_chain_id ASC
+"#
+            }
+        };
+
+        return (sql.to_string(), vec![Value::BigInt(Some(chain_id))]);
```
[the remark this answers is not in the document]
Comment (EvgenKor) [reply interchain-indexer/interchain-indexer-logic/src/database.rs:224]: Good point. Fixed in https://github.com/blockscout/blockscout-rs/pull/1662/commits/3a445f9e498cd928c8446298e732abae80949629
